From d44c8de93fc9ba8c0585b035b69359a2304ed90e Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Wed, 2 Aug 2017 14:48:29 +0800
Subject: [PATCH 2278/5242] MLK-16119-03: dts: Enable hdmi driver to imx8mq
 evk dts

commit  b98cc2d0b7ec002cd993795bd3fea849fde4d3a3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add hdmi driver property to imx8mq evk.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |    5 +++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi    |   11 +++++++++++
 2 files changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 7bd7788..6a9642f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -405,6 +405,11 @@
 	};
 };
 
+&hdmi {
+	disp-dev = "hdmi_disp";
+	status = "okay";
+};
+
 &i2c2 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 3861e06..f8234f4 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -870,6 +870,17 @@
 		only-dma-mask32 = <1>;
 	};
 
+	hdmi: hdmi@32c00000 {
+		compatible = "fsl,imx8mq-hdmi";
+		reg = <0x0 0x32c00000 0x0 0x100000>,	/* HDMI registers */
+				<0x0 0x32e40000 0x0 0x40000>;   /* HDMI SEC register */
+		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
+		video-mode = <95>;	/* #16: 1920x1080p@60HZ 16:9 */
+							/* #95: 3840x2160p@30Hz 16:9 */
+							/* #99: 3840x2160p@60Hz 16:9 */
+		status = "disabled";
+	};
+
 	pcie0: pcie@0x33800000 {
 		compatible = "fsl,imx8mq-pcie", "snps,dw-pcie";
 		reg = <0x0 0x33800000 0x0 0x400000>, <0x0 0x1ff00000 0x0 0x80000>;
-- 
1.7.9.5

